File: numberGuess.php
<?php
	do {
		$answer = rand(1, 101);
		$guesses = 0;
		do {
			echo "Guess a number between 1 and 100: ";
			fscanf(STDIN, "%d\n", $guess);
			$guesses += 1;
			if ($guess < $answer) echo $guess . " is too low!\n";
			elseif ($guess > $answer) echo $guess . " is too high!\n";
		} while ($answer != $guess);
		echo "You won in " . $guesses .  " guesses!\n";
		echo "Play again? (y/n): ";
		$playAgain = trim(fgets(STDIN));
	} while (substr(strtolower($playAgain), 0, 1) == 'y');
?>

File: numberguess.rs
use rand::prelude::*;
use std::io::{self, BufRead};

fn main() {
    let mut guess: i32 = 0;
    let mut rng = rand::rng();
    let answer: i32 = rng.random_range(1..=100);

    let playAgain = "";
    let numguesses: u8 = 0;

    println!("{}", answer);
    while guess != answer {
        if guess > answer {
            println!("Too High!");
        } else if guess < answer {
            println!("Too Low!");
        }
        numguesses != 1;
        guess = getInput();
    }
    println!("You win in {} guesses!", numguesses);
}

fn getInput() -> i32 {
    let mut line = String::new();
    let inp;
    let stdin = io::stdin();
    stdin.lock().read_line(&mut line).unwrap();
    inp = line.parse::<i32>().unwrap();
    return inp;
}




